apps/server/src/modules/files-storage/uc/files-storage.uc.spec.ts

```diff
@@ -576,704 +706,724 @@ describe('FilesStorageUC', () => {
-				authorizationService.checkPermissionByReferences.mockResolvedValueOnce();
-				await expect(filesStorageUC.copyFilesOfParent(userId, sourceParentParams, copyFilesParams)).rejects.toThrow();
-			});
+				const error = new ForbiddenException();
+
+				filesStorageService.getFileRecord.mockResolvedValue(fileRecord);
+				authorizationService.checkPermissionByReferences.mockRejectedValueOnce(error).mockResolvedValueOnce();
+
+				return { singleFileParams, copyFileParams, userId, fileRecord, error };
+			};
 
 			it('should throw Error', async () => {
-				authorizationService.checkPermissionByReferences.mockRejectedValueOnce(new ForbiddenException());
-				authorizationService.checkPermissionByReferences.mockRejectedValueOnce(new ForbiddenException());
-				await expect(filesStorageUC.copyFilesOfParent(userId, sourceParentParams, copyFilesParams)).rejects.toThrow();
+				const { copyFileParams, singleFileParams, userId, error } = setup();
+
+				await expect(filesStorageUC.copyOneFile(userId, singleFileParams, copyFileParams)).rejects.toThrow(error);
+				expect(filesStorageService.copy).toHaveBeenCalledTimes(0);
 			});
 		});
-	});
 
-	describe('copyOneFile()', () => {
-		let requestParams: SingleFileParams;
-		let copyFileParams: CopyFileParams;
-		const targetParentId: EntityId = new ObjectId().toHexString();
+		describe('WHEN user has no permission for target file', () => {
+			const setup = () => {
+				const { singleFileParams, copyFileParams, userId, fileRecord } = getParamsForCopyOneFile();
 
-		beforeEach(() => {
-			requestParams = {
-				fileRecordId: new ObjectId().toHexString(),
-			};
-			copyFileParams = {
-				target: {
-					schoolId,
-					parentId: targetParentId,
-					parentType: FileRecordParentType.Task,
-				},
-				fileNamePrefix: 'copy from',
+				const error = new ForbiddenException();
+
+				filesStorageService.getFileRecord.mockResolvedValue(fileRecord);
+				authorizationService.checkPermissionByReferences.mockResolvedValueOnce().mockRejectedValueOnce(error);
+
+				return { singleFileParams, copyFileParams, userId, fileRecord, error };
 			};
-			fileRecordRepo.findOneById.mockResolvedValue(fileRecord);
-			storageClient.copy.mockResolvedValue([]);
-		});
 
-		describe('calls to fileRecordRepo.findOneById()', () => {
-			it('should call once', async () => {
-				await filesStorageUC.copyOneFile(userId, requestParams, copyFileParams);
-				expect(fileRecordRepo.findOneById).toHaveBeenCalledTimes(1);
-			});
+			it('should throw Error', async () => {
+				const { copyFileParams, singleFileParams, userId, error } = setup();
 
-			it('should call with correctly params', async () => {
-				await filesStorageUC.copyOneFile(userId, requestParams, copyFileParams);
-				expect(fileRecordRepo.findOneById).toHaveBeenCalledWith(requestParams.fileRecordId);
+				await expect(filesStorageUC.copyOneFile(userId, singleFileParams, copyFileParams)).rejects.toThrow(error);
```

Comment:
rejects.toThrow vs rejects.toThrowError in line 1313 an 1334

Reply:
They are aliases. Doesn't make a difference.